tests/unit/test_dotenvpath.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+
+import pytest
+
+import ni_measurementlink_service
+from ni_measurementlink_service import _dotenvpath
+
+
+@pytest.mark.parametrize("dotenv_exists", [False, True])
+def test___dotenv_exists_varies___has_dotenv_file___matches_dotenv_exists(
+    dotenv_exists: bool, tmp_path: Path
+) -> None:
+    if dotenv_exists:
+        (tmp_path / ".env").write_text("")
+    subdirs = [tmp_path / "a", tmp_path / "a" / "b", tmp_path / "a" / "b" / "c"]
+    for dir in subdirs:
+        dir.mkdir()
+
+    assert _dotenvpath._has_dotenv_file(tmp_path) == dotenv_exists
+    assert all([_dotenvpath._has_dotenv_file(p) == dotenv_exists for p in subdirs])
+
+
+def test___get_caller_path___returns_this_modules_path() -> None:
+    assert _dotenvpath._get_caller_path() == Path(__file__)
```

Comment:
Windows with Python 3.11:
```
FAILED tests/unit/test_dotenvpath.py::test___get_caller_path___returns_this_modules_path - AssertionError: assert WindowsPath('D:/a/measurementlink-python/measurementlink-python/.venv/Lib/site-packages/pluggy/_callers.py') == WindowsPath('D:/a/measurementlink-python/measurementlink-python/tests/unit/test_dotenvpath.py')
 +  where WindowsPath('D:/a/measurementlink-python/measurementlink-python/.venv/Lib/site-packages/pluggy/_callers.py') = <function _get_caller_path at 0x00000268B012C0E0>()
 +    where <function _get_caller_path at 0x00000268B012C0E0> = _dotenvpath._get_caller_path
 +  and   WindowsPath('D:/a/measurementlink-python/measurementlink-python/tests/unit/test_dotenvpath.py') = Path('D:\\a\\measurementlink-python\\measurementlink-python\\tests\\unit\\test_dotenvpath.py')
```

Reply:
In Python 3.11 and later, `traceback.walk_stack(None)` skips more stack frames than in the past: https://github.com/python/cpython/issues/96092